### ISSUE TITLE
Check out NURBS-Diff

### ISSUE THREAD

**Issue by mcw92** [ARTIST-Association/ARTIST#13]
There is another implementation of differentiable NURBS out there, providing a torch layer for NURBS evaluation for given control points and parametrization:

> In this paper, we have developed a differentiable NURBS module that can be used in machine learning and CAD applications. The key contributions of this work are:
> 1. A differentiable programming framework using NURBS representation where the losses can be back-propagated using the NURBS definitions. Specifically, we define the Jacobian based on the derivatives of the NURBS with respect to its input parameters.
> 2. A GPU-accelerated implementation of our NURBS module in PyTorch for better integration with existing deep learning programming frameworks.
> 3. A gradient descent-based optimization framework using NURBS-Diff for performing CAD operations such as curve or surface fitting and surface offsetting.
> 4. The applicability of our proposed differentiable programming framework to extend the training process of unsupervised point cloud reconstruction of NURBS surfaces.

Repo: https://github.com/idealab-isu/NURBSDiff
Paper: https://doi.org/10.1016/j.cad.2022.103199

Note that this also depends on Pytorch3D.
